File: 151002/Protchenko/lab45/publisher/src/main/java/by/bsuir/publisher/model/mapper/IssueMapper.java
package by.bsuir.publisher.model.mapper;

import by.bsuir.publisher.model.dto.request.IssueRequestDto;
import by.bsuir.publisher.model.dto.response.IssueResponseDto;
import by.bsuir.publisher.model.entity.Issue;
import org.mapstruct.*;

import java.util.List;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING, uses = CustomMapper.class)
public interface IssueMapper {

    @Mapping(target = "authorId", source = "author.id")
    IssueResponseDto toDto(Issue Issue);

    @Mapping(target = "created", ignore = true)
    @Mapping(target = "modified", ignore = true)
    @Mapping(target = "author", source = "authorId", qualifiedByName = "authorIdToAuthorRef")
    Issue toEntity(IssueRequestDto IssueRequestDto);

    List<IssueResponseDto> toDto(List<Issue> Issues);

    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
    @Mapping(target = "id", ignore = true)
    @Mapping(target = "author", source = "authorId", qualifiedByName = "authorIdToAuthorRef")
    Issue partialUpdate(IssueRequestDto tweetRequestDto, @MappingTarget Issue issue);
}

File: 151003/Barylko/NoSQL/Module.Publisher/Mappers/IssueMapper.cs
using Publisher.Dto.Request.CreateTo;
using Publisher.Dto.Response;
using Publisher.Models;
using Riok.Mapperly.Abstractions;
namespace Publisher.Mappers;

[Mapper]
public static partial class IssueMapper
{
    public static partial Issue ToEntity(this CreateIssueRequestTo createIssueRequestTo);
    public static partial IssueResponseTo ToResponse(this Issue Issue);
    public static partial IEnumerable<IssueResponseTo> ToResponse(this IEnumerable<Issue> Issues);
}